plugin/core/edit.py

```diff
@@ -55,41 +56,29 @@ def open_and_apply_edits(self, path, file_changes):
 
 class LspApplyDocumentEditCommand(sublime_plugin.TextCommand):
     def run(self, edit, changes: 'Optional[List[dict]]' = None, show_status=True):
+        # Apply the changes in reverse, so that we don't invalidate the range
+        # of any change that we haven't applied yet.
+        start = timer()
+        changes2 = changes or []
+        indices = self.changes_order(changes2)
+        for index in indices:
+            change = changes2[index]
+            self.apply_change(self.create_region(change), change.get('newText'), edit)
+        elapsed = timer() - start
 
-        # Sort changes due to issues with self.view.get_regions
-        # See https://github.com/tomv564/LSP/issues/325
-        changes = self.changes_sorted(changes) if changes else []
-        regions = list(self.create_region(change) for change in changes)
-        replacements = list(change.get('newText') for change in changes)
-
-        # TODO why source.python here?
-        self.view.add_regions('lsp_edit', regions, "source.python")
-
-        index = 0
-        last_region_count = len(regions)
-        for newText in replacements:
-            # refresh updated regions after each edit.
-            updated_regions = self.view.get_regions('lsp_edit')
-            region = updated_regions[index]  #
-            self.apply_change(region, newText, edit)
-            if len(self.view.get_regions('lsp_edit')) == last_region_count and last_region_count > 1:
-                index += 1  # no regions lost, move to next region.
-            else:
-                # current region was removed, don't advance index.
-                last_region_count = len(self.view.get_regions('lsp_edit'))
-
-        self.view.erase_regions('lsp_edit')
         if show_status:
             window = self.view.window()
             if window:
                 base_dir = get_project_path(window)
                 file_path = self.view.file_name()
                 relative_file_path = os.path.relpath(file_path, base_dir) if base_dir else file_path
-                message = 'Applied {} change(s) to {}'.format(len(changes), relative_file_path)
+                message = 'Applied {} change(s) to {} in {:.1f} ms'.format(
+                    len(indices), relative_file_path, elapsed * 1000)
                 window.status_message(message)
+                debug(message)
```

Comment:
the extra `debug(message)` here feels unnecessary.

Reply:
I added it because the window.status_message disappears after a second or so, and I often missed it.